api/src/pcapi/core/subscription/phone_validation/api.py

```diff
@@ -169,7 +172,12 @@ def validate_phone_number(user: users_models.User, code: str) -> None:
     _check_and_update_phone_validation_attempts(app.redis_client, user)
 
     try:
-        token = token_utils.SixDigitsToken.load_and_check(code, token_utils.TokenType.PHONE_VALIDATION, user.id)
+        if settings.ENABLE_PHONE_VALIDATION_BYPASS and PHONE_VALIDATION_TEST_EMAIL_RE.match(user.email):
+            token = token_utils.SixDigitsToken.load_without_checking(
+                code, token_utils.TokenType.PHONE_VALIDATION, user.id
+            )
+        else:
+            token = token_utils.SixDigitsToken.load_and_check(code, token_utils.TokenType.PHONE_VALIDATION, user.id)
```

Comment:
je ne sais pas si c'est mieux

les fonctions ayant la meme interface, on pourrait juste changer la fonction qui est appelé

```suggestion
        load = (
            token_utils.SixDigitsToken.load_without_checking
            if settings.ENABLE_PHONE_VALIDATION_BYPASS and PHONE_VALIDATION_TEST_EMAIL_RE.match(user.email)
            else token_utils.SixDigitsToken.load_and_check
        )
        token = load(
            code, token_utils.TokenType.PHONE_VALIDATION, user.id
        )
```

Reply:
Pourquoi pas ... Ça peut faire plus propre oui.

Reply:
Tu peux aussi déporter la logique directement dans `load_and_check` de `SixDigitsToken`
Comme ça pour le reste du code c'est simple plus à lire

Reply:
Intéressant comme idée, je vais regarder si c'est facilement faisable :)

Reply:
Je ne pense pas déporter la logique dans le `load_and_check` de `SixDigitsToken` parce que je ne pense pas que ce soit le rôle de cette classe de gérer le cas des tests end-to-end pour la validation du téléphone de l'utilisateur, on pourrait avoir un `SixDigitsToken` utilisé pour un autre cas. De plus cette classe n'a pas l'email de l'utilisateur.